build_runner_core/lib/src/changes/build_script_updates.dart

```diff
@@ -52,8 +52,13 @@ class BuildScriptUpdates {
     return BuildScriptUpdates._(supportsIncrementalRebuilds, allSources);
   }
 
-  static Iterable<Uri> get _urisForThisScript =>
-      currentMirrorSystem().libraries.keys;
+  static Set<Uri> get _urisForThisScript => currentMirrorSystem()
```

Comment:
any idea if this visibly impacts performance? I would guess not - even though it could potentially do a lot more work in practice there shouldn't be to many top level declarations in each library...